cmd/size_reservations.go

```diff
@@ -0,0 +1,235 @@
+			if err != nil {
+				return nil, err
+			}
+
+			return &models.V1SizeReservationUpdateRequest{ //nolint:exhaustruct
+				Amount:       pointer.PointerOrNil(int32(viper.GetInt32("amount"))),
+				Description:  viper.GetString("description"),
+				ID:           &id,
+				Labels:       labels,
+				Partitionids: viper.GetStringSlice("partitions"),
+			}, nil
+		},
+		UpdateCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.Flags().Int32("amount", 0, "the amount to associate with this reservation")
+			cmd.Flags().StringSlice("partitions", nil, "the partition ids to associate with this reservation")
+			cmd.Flags().StringSlice("labels", nil, "the labels to associate with this reservation")
+			cmd.Flags().String("description", "", "the description to associate with this reservation")
+
+			genericcli.Must(cmd.RegisterFlagCompletionFunc("partitions", c.comp.PartitionListCompletion))
+		},
+		ListCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.Flags().String("id", "", "the id to filter")
+			cmd.Flags().String("size", "", "the size id to filter")
+			cmd.Flags().String("project", "", "the project id to filter")
+			cmd.Flags().String("partition", "", "the partition id to filter")
+
+			genericcli.Must(cmd.RegisterFlagCompletionFunc("id", c.comp.SizeReservationsListCompletion))
+			genericcli.Must(cmd.RegisterFlagCompletionFunc("size", c.comp.SizeListCompletion))
+			genericcli.Must(cmd.RegisterFlagCompletionFunc("project", c.comp.ProjectListCompletion))
+			genericcli.Must(cmd.RegisterFlagCompletionFunc("partition", c.comp.PartitionListCompletion))
+		},
+	}
+
+	usageCmd := &cobra.Command{
+		Use:   "usage",
+		Short: "see current usage of size reservations",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return w.usage()
+		},
+	}
+
+	usageCmd.Flags().String("size-id", "", "the size-id to filter")
+	usageCmd.Flags().String("project", "", "the project to filter")
+	usageCmd.Flags().String("partition", "", "the partition to filter")
+
+	genericcli.Must(usageCmd.RegisterFlagCompletionFunc("size-id", c.comp.SizeListCompletion))
+	genericcli.Must(usageCmd.RegisterFlagCompletionFunc("project", c.comp.ProjectListCompletion))
+	genericcli.Must(usageCmd.RegisterFlagCompletionFunc("partition", c.comp.PartitionListCompletion))
+
+	genericcli.AddSortFlag(usageCmd, sorters.SizeReservationsUsageSorter())
+
+	return genericcli.NewCmds(cmdsConfig, usageCmd)
+}
+
+func (c sizeReservationsCmd) Get(id string) (*models.V1SizeReservationResponse, error) {
+	resp, err := c.client.Size().GetSizeReservation(sizemodel.NewGetSizeReservationParams().WithID(id), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.Payload, nil
+}
+
+func (c sizeReservationsCmd) List() ([]*models.V1SizeReservationResponse, error) {
+	resp, err := c.client.Size().FindSizeReservations(sizemodel.NewFindSizeReservationsParams().WithBody(&models.V1SizeReservationListRequest{
+		ID:          viper.GetString("id"),
+		Partitionid: viper.GetString("parition"),
```

Comment:
Typo. Should be `partition`.

Reply:
Should be fixed with: https://github.com/metal-stack/metalctl/pull/265